Add basic union type support to property-info php doc extractor

temp opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no 
| Tickets       | Fix #45903
| License       | MIT
| Doc PR        | 

This PR adds basic union type support to PropertyInfo / PhpDocExtractor. Unit tests are copied from PhpStanExtractor, which already handles union types. Not all test cases can be fulfilled by the PhpDocExtractor, though. These tests are disabled.

OskarStark commented on src/Symfony/Component/PropertyInfo/Tests/Extractor/PhpDocExtractorTest.php:
```diff
@@ -409,6 +410,27 @@ public function constructorTypesProvider()
+            ['c', [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, [new Type(Type::BUILTIN_TYPE_INT)], [new Type(Type::BUILTIN_TYPE_STRING), new Type(Type::BUILTIN_TYPE_INT)])]],
+            ['d', null], // union keys are not supported by phpdocumentor/reflection
+            ['e', null], // union keys are not supported by phpdocumentor/reflection
+            // ['f', null], // constant expressions are not supported
```
Should be removed then

temp replied:
This was just for discussion, i'll remove it.

OskarStark replied:
Ah ok, yes I read it.

Lets keep it for now

temp replied:
Damn, I just pushed. Should I restore it? Or is the notice here enough?

OskarStark commented:
As this is a new feature, it should be against `6.3` branch, thank you

temp commented:
> As this is a new feature, it should be against `6.3` branch, thank you

Looks like I can't change the target branch?

temp commented:
> As this is a new feature, it should be against `6.3` branch, thank you

No 6.3 branch yet? Changed it to 6.2 for now.

alexandre-daubois committed:
[VarDumper] Fix typo in deprecation message

nicolas-grekas committed:
minor #48363 [VarDumper] Fix typo in deprecation message (alexandre-daubois)

This PR was merged into the 6.2 branch.

Discussion
----------

[VarDumper] Fix typo in deprecation message

| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | _N/A_
| License       | MIT
| Doc PR        | _N/A_

Commits
-------

dc22f2ac16 [VarDumper] Fix typo in deprecation message

fabpot committed:
[Mailer] Fix body renderer check

fabpot committed:
bug #48366 [Mailer] Fix body renderer check (fabpot)

This PR was merged into the 6.2 branch.

Discussion
----------

[Mailer] Fix body renderer check

| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | yes
| New feature?  | no <!-- please update src/**/CHANGELOG.md files -->
| Deprecations? | no <!-- please update UPGRADE-*.md and src/**/CHANGELOG.md files -->
| Tickets       | n/a <!-- prefix each issue number with "Fix #", no need to create an issue if none exists, explain below instead -->
| License       | MIT
| Doc PR        | n/a

What I did in #47049 does not work as one can have many message listeners. So, we need to do this check after all listeners are run.

Commits
-------

8bb752265a [Mailer] Fix body renderer check

fabpot committed:
Update CHANGELOG for 6.2.0-RC2

fabpot committed:
Update VERSION for 6.2.0-RC2

fabpot committed:
Merge pull request #48372 from fabpot/release-6.2.0-RC2

released v6.2.0-RC2

fabpot committed:
Bump Symfony version to 6.2.0
